links/link_vxlan.go

```diff
@@ -0,0 +1,368 @@
+	// this will be used for the vxlan and and veth endpoint on the host side
+	// but with different prefixes
+	if len(ifaceNamePost) > 14 {
+		oldName := ifaceNamePost
+		ifaceNamePost = stableHashedInterfacename(ifaceNamePost, 8)
+		log.Debugf("can't use %s as interface name postfix, falling back to %s", oldName, ifaceNamePost)
+	}
+
+	// prepare the vxlan struct
+	vxlanLink, err := lr.resolveStitchedVxlan(params, ifaceNamePost)
+	if err != nil {
+		return nil, err
+	}
+
+	// prepare the veth struct
+	vethLink, stitchEp, err := lr.resolveStitchedVEth(params, ifaceNamePost)
+	if err != nil {
+		return nil, err
+	}
+
+	// return the stitched vxlan link
+	stitchedLink := NewVxlanStitched(vxlanLink, vethLink, stitchEp)
+
+	// add stitched link to node
+	params.Nodes[lr.Endpoint.Node].AddLink(stitchedLink)
+
+	return stitchedLink, nil
+}
+
+func (lr *LinkVxlanRaw) resolveRegular(params *ResolveParams) (Link, error) {
+	var err error
+	link := &LinkVxlan{
+		LinkCommonParams: lr.LinkCommonParams,
+		noLearning:       lr.NoLearning,
+		noL2Miss:         lr.NoL2Miss,
+		noL3Miss:         lr.NoL3Miss,
+	}
+
+	// resolve local Endpoint
+	link.localEndpoint, err = lr.Endpoint.Resolve(params, link)
+	if err != nil {
+		return nil, err
+	}
+
+	ip := net.ParseIP(lr.Remote)
+	// if the returned ip is nil, an error occured.
+	// we consider, that we maybe have a textual hostname
+	// e.g. dns name so we try to resolve the string next
+	if ip == nil {
+		ips, err := net.LookupIP(lr.Remote)
+		if err != nil {
+			return nil, err
+		}
+
+		// prepare log message
+		sb := strings.Builder{}
+		for _, ip := range ips {
+			sb.WriteString(", ")
+			sb.WriteString(ip.String())
+		}
+		log.Debugf("looked up hostname %s, received IP addresses [%s]", lr.Remote, sb.String()[2:])
+
+		// always use the first address
+		if len(ips) <= 0 {
+			return nil, fmt.Errorf("unable to resolve %s", lr.Remote)
+		}
+		ip = ips[0]
+	}
+
+	parentIf := lr.ParentInterface
+
+	if parentIf == "" {
+		conn, err := rtnl.Dial(nil)
+		if err != nil {
+			return nil, fmt.Errorf("can't establish netlink connection: %s", err)
+		}
+		defer conn.Close()
+		r, err := conn.RouteGet(ip)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find a route to VxLAN remote address %s", ip.String())
+		}
+		parentIf = r.Interface.Name
+	}
+
+	// resolve remote endpoint
+	link.remoteEndpoint = NewEndpointVxlan(params.Nodes["host"], link)
+	link.remoteEndpoint.parentIface = parentIf
+	link.remoteEndpoint.udpPort = lr.UdpPort
+	link.remoteEndpoint.remote = ip
+	link.remoteEndpoint.vni = lr.Vni
+
+	// add link to local endpoints node
+	link.localEndpoint.GetNode().AddLink(link)
+
+	return link, nil
+}
+
+func (*LinkVxlanRaw) GetType() LinkType {
+	return LinkTypeVxlan
+}
+
+type LinkVxlan struct {
+	LinkCommonParams
+	localEndpoint  Endpoint
+	remoteEndpoint *EndpointVxlan
+	noLearning     bool
+	noL2Miss       bool
+	noL3Miss       bool
+}
```

Comment:
I don't think it is good to have localEndpoint and remoteEndpoint
Vxlan link seems to be best to be described with a single endpoint which is the link itself.
We create a single endpoint and put it into the node's netns

Reply:
removed them [f7965ff](https://github.com/srl-labs/containerlab/pull/1532/commits/f7965ffa21b298273d07028a3e4ded00a0d02144)